client/widgets/TextControls.cpp

```diff
@@ -189,8 +200,26 @@ void CTextContainer::blitLine(Canvas & to, Rect destRect, std::string what)
 		{
 			std::string toPrint = what.substr(begin, end - begin);
 
-			if(currDelimeter % 2) // Enclosed in {} text - set to yellow
-				to.drawText(where, font, Colors::YELLOW, ETextAlignment::TOPLEFT, toPrint);
+			if(currDelimeter % 2) // Enclosed in {} text - set to yellow or defined color
+			{
+				std::smatch match;
+   				std::regex expr("^(.*?)\\|");
+				if(std::regex_search(toPrint, match, expr))
+				{
+					std::string colorText = match[1].str();
+					ColorRGBA color = CMessage::parseColor(colorText);
+					
+					if(color.a != Colors::TRANSPARENCY.ALPHA_TRANSPARENT)
```

Comment:
Is this to handle parsing error? Probably unnecessary. If color is broken it is up to author of this text to fix, not vcmi

Reply:
See above comment. If no color is detected, then the syntax is ignored.